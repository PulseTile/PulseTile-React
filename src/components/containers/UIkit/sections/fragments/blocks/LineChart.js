import React from 'react';
import { get } from 'lodash';
import { Line } from 'react-chartjs-2';

const dataChartLine = {
  labels: ["03-Aug-15 01:11","01:23","01:23","01:23","01:23","01:23","01:23","06:11","06:23","06:23"],
  datasets:[
    { label: "Resp", backgroundColor: "rgba(13, 141, 5, 0.4)",   borderColor:"rgba(13, 141, 5, 1)",   pointBorderColor: "rgba(13, 141, 5, 1)",   pointBackgroundColor:"rgba(13, 141, 5, 1)",   data: [25,23,23,23,23,23,23,25,23,23] },
    { label: "SpO2", backgroundColor: "rgba(219, 0, 120, 0.4)",  borderColor:"rgba(219, 0, 120, 1)",  pointBorderColor: "rgba(219, 0, 120, 1)",  pointBackgroundColor:"rgba(219, 0, 120, 1)",  data: [97,97,97,97,97,97,97,97,97,97] },
    { label: "HR",   backgroundColor: "rgba(70, 124, 174, 0.4)", borderColor:"rgba(70, 124, 174, 1)", pointBorderColor: "rgba(70, 124, 174, 1)", pointBackgroundColor:"rgba(70, 124, 174, 1)", data: [45,45,45,45,45,45,45,45,45,45] },
    { label: "SBP",  backgroundColor: "rgba(236, 109, 28, 0.4)", borderColor:"rgba(236, 109, 28, 1)", pointBorderColor: "rgba(236, 109, 28, 1)", pointBackgroundColor:"rgba(236, 109, 28, 1)", data: [90,92,92,92,92,92,92,90,92,92] },
    { label: "DBP",  backgroundColor: "rgba(5, 186, 195, 0.4)",  borderColor:"rgba(5, 186, 195, 1)",  pointBorderColor: "rgba(5, 186, 195, 1)",  pointBackgroundColor:"rgba(5, 186, 195, 1)",  data: [60,64,64,64,64,64,64,60,64,64] },
    { label: "Temp", backgroundColor: "rgba(221, 43, 8, 0.4)",   borderColor:"rgba(221, 43, 8, 1)",   pointBorderColor: "rgba(221, 43, 8, 1)",   pointBackgroundColor:"rgba(221, 43, 8, 1)",   data: [60,64,64,64,64,64,64,60,64,64] }
  ]
};

/**
 * This component returns content of LineChart section in Blocks
 */
const LineChart = () => {
  return (
    <div className="ui-sub-section">
      <strong className="ui-sub-title">Chart Lines</strong>
      <div className="wrap-chart chart-vitals bordered">
        <Line
          data={{
              labels: dataChartLine.labels,
              datasets: dataChartLine.datasets
          }}
          onElementsClick={function () {}}
          width={600}
          height={441}
          options={{
              capBezierPoints: false,
              responsive: true,
              maintainAspectRatio: false,
              elements: {
                line: {
                  fill: false,
                  lineTension: 0,
                  borderCapStyle: 'butt',
                  borderDashOffset: 0.0,
                  borderJoinStyle: 'miter',
                  pointBorderWidth: 5,
                  pointRadius: 1,
                  pointHoverBorderWidth: 8,
                  pointHoverRadius: 1,
                  pointHitRadius: 8,
                  spanGaps: true,
                },
              },
              tooltips: {
                mode: 'label',
                titleMarginBottom: 15,
                bodySpacing: 10,
                xPadding: 10,
                yPadding: 10,
                caretSize: 0,
                callbacks: {
                  label(tooltipItem, data) {
                    return `  ${data.datasets[tooltipItem.datasetIndex].label} : ${tooltipItem.yLabel}`;
                  },
                },
              },
              scales: {
                xAxes: [{
                    ticks: {
                      maxRotation: 90,
                      minRotation: 90,
                    },
                  }
                ],
              },
              layout: {
                padding: {
                  right: 10,
                },
              },
            }
          }
        />
      </div>
    </div>
  );
};

export default LineChart;
