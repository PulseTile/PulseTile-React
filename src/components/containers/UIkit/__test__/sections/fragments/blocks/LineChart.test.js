import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import LineChart from '../../../../sections/fragments/blocks/LineChart';

Enzyme.configure({ adapter: new Adapter() });

describe('Component <LineChart />', () => {
  it('should renders LineChart with props correctly', () => {
    const component = shallow(<LineChart />);
    expect(component).toMatchSnapshot();
  });
});
