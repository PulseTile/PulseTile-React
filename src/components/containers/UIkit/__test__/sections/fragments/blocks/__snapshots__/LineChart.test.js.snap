// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <LineChart /> should renders LineChart with props correctly 1`] = `
<div
  className="ui-sub-section"
>
  <strong
    className="ui-sub-title"
  >
    Chart Lines
  </strong>
  <div
    className="wrap-chart chart-vitals bordered"
  >
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "rgba(13, 141, 5, 0.4)",
              "borderColor": "rgba(13, 141, 5, 1)",
              "data": Array [
                25,
                23,
                23,
                23,
                23,
                23,
                23,
                25,
                23,
                23,
              ],
              "label": "Resp",
              "pointBackgroundColor": "rgba(13, 141, 5, 1)",
              "pointBorderColor": "rgba(13, 141, 5, 1)",
            },
            Object {
              "backgroundColor": "rgba(219, 0, 120, 0.4)",
              "borderColor": "rgba(219, 0, 120, 1)",
              "data": Array [
                97,
                97,
                97,
                97,
                97,
                97,
                97,
                97,
                97,
                97,
              ],
              "label": "SpO2",
              "pointBackgroundColor": "rgba(219, 0, 120, 1)",
              "pointBorderColor": "rgba(219, 0, 120, 1)",
            },
            Object {
              "backgroundColor": "rgba(70, 124, 174, 0.4)",
              "borderColor": "rgba(70, 124, 174, 1)",
              "data": Array [
                45,
                45,
                45,
                45,
                45,
                45,
                45,
                45,
                45,
                45,
              ],
              "label": "HR",
              "pointBackgroundColor": "rgba(70, 124, 174, 1)",
              "pointBorderColor": "rgba(70, 124, 174, 1)",
            },
            Object {
              "backgroundColor": "rgba(236, 109, 28, 0.4)",
              "borderColor": "rgba(236, 109, 28, 1)",
              "data": Array [
                90,
                92,
                92,
                92,
                92,
                92,
                92,
                90,
                92,
                92,
              ],
              "label": "SBP",
              "pointBackgroundColor": "rgba(236, 109, 28, 1)",
              "pointBorderColor": "rgba(236, 109, 28, 1)",
            },
            Object {
              "backgroundColor": "rgba(5, 186, 195, 0.4)",
              "borderColor": "rgba(5, 186, 195, 1)",
              "data": Array [
                60,
                64,
                64,
                64,
                64,
                64,
                64,
                60,
                64,
                64,
              ],
              "label": "DBP",
              "pointBackgroundColor": "rgba(5, 186, 195, 1)",
              "pointBorderColor": "rgba(5, 186, 195, 1)",
            },
            Object {
              "backgroundColor": "rgba(221, 43, 8, 0.4)",
              "borderColor": "rgba(221, 43, 8, 1)",
              "data": Array [
                60,
                64,
                64,
                64,
                64,
                64,
                64,
                60,
                64,
                64,
              ],
              "label": "Temp",
              "pointBackgroundColor": "rgba(221, 43, 8, 1)",
              "pointBorderColor": "rgba(221, 43, 8, 1)",
            },
          ],
          "labels": Array [
            "03-Aug-15 01:11",
            "01:23",
            "01:23",
            "01:23",
            "01:23",
            "01:23",
            "01:23",
            "06:11",
            "06:23",
            "06:23",
          ],
        }
      }
      height={441}
      onElementsClick={[Function]}
      options={
        Object {
          "capBezierPoints": false,
          "elements": Object {
            "line": Object {
              "borderCapStyle": "butt",
              "borderDashOffset": 0,
              "borderJoinStyle": "miter",
              "fill": false,
              "lineTension": 0,
              "pointBorderWidth": 5,
              "pointHitRadius": 8,
              "pointHoverBorderWidth": 8,
              "pointHoverRadius": 1,
              "pointRadius": 1,
              "spanGaps": true,
            },
          },
          "layout": Object {
            "padding": Object {
              "right": 10,
            },
          },
          "maintainAspectRatio": false,
          "responsive": true,
          "scales": Object {
            "xAxes": Array [
              Object {
                "ticks": Object {
                  "maxRotation": 90,
                  "minRotation": 90,
                },
              },
            ],
          },
          "tooltips": Object {
            "bodySpacing": 10,
            "callbacks": Object {
              "label": [Function],
            },
            "caretSize": 0,
            "mode": "label",
            "titleMarginBottom": 15,
            "xPadding": 10,
            "yPadding": 10,
          },
        }
      }
      width={600}
    />
  </div>
</div>
`;
